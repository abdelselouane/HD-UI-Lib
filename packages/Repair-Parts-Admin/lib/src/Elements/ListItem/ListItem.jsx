import React from 'react';
import PropTypes from 'prop-types';

const ListItem = (props) => {

  const {
    title,
    data,
    actions,
    columns
  } = props;
  
  return (
    <div
      className="card"
      style={{marginTop: "20px", width: "100%"}}>
      <div className="card-body">
        <h2
          className="card-title"
          style={{color:"#f96302"}}
        >{title}{data.partNbr}</h2>
        <div className="row">
          {
            columns.map( (column, index) => {
              return (<div
                className={column.className}
                style={ column.actions && {textAlign: 'right'} }
                key={index} >
                { column.label &&
                <label>{column.label}</label>
                }
                { data[column.key] &&
                <div><strong>{column.key === 'cost' && '$'}{data[column.key]}</strong></div>
                }
                {
                  column.actions && actions.map( (action, index) => {
                    return (<button
                      className={action.className}
                      id={`part-${data.partNbr}`}
                      key={index}
                      onClick={action.onClick} >
                      {action.text}
                    </button>);
                  })
                }
              </div>);
            })
          }
        </div>
      </div>
    </div>
  );
};

ListItem.propTypes = {
  title: PropTypes.string.isRequired,
  data: PropTypes.object.isRequired,
  actions: PropTypes.array,
  columns: PropTypes.array
};

export default ListItem;