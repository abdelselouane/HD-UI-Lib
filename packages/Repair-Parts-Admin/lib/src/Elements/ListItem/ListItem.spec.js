import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import ListItem from './ListItem';
import toJson from 'enzyme-to-json';
import * as Enzyme from 'enzyme';
import ReactSixteenAdapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new ReactSixteenAdapter() });

describe('<ListItem />', () => {
    let wrapper;
    let props;
    let parts;

    afterEach(() => {
        jest.clearAllMocks();
    });

    beforeAll(() => {
        ReactDOM.createPortal = jest.fn();

        parts = {
            partNbr: 'mockPartNumber1',
            partDescription: 'mockDescription1',
            brandName: 'mockBrandName1',
            supplierName: 'mockSupplierName1',
            cost: 10
        };

        props = {
            title: "Part #",
            columns: [{
                className: "col-lg-2",
                label: "Brand",
                key: "brandName"
            },
            {
                className: "col-lg-3",
                label: "Description",
                key: "partDesc"
            },
            {
                className: "col-lg-2",
                label: "Supplier",
                key: "supplierName"
            },
            {
                className: "col-lg-2",
                label: "Cost",
                key: "cost"
            },
            {
                className: "col-lg-3",
                actions: true
            }],
            actions: [{
                text: "Edit Part",
                className: "button md",
                onClick: jest.fn()
            }]
        };
        wrapper = shallow(<ListItem {...props} data={parts} />);
    });

    it('should render properly', () => {
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('should call handleEditPart on Edit Part Button', () => {
        const spy = jest.spyOn(props.actions[0], 'onClick');
        const listButton = wrapper.find('.button');
        listButton.simulate('click');
        expect(spy).toHaveBeenCalled();
    });

});