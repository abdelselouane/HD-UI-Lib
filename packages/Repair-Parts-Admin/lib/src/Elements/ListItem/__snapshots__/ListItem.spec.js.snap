// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ListItem /> should render properly 1`] = `
<div
  className="card"
  style={
    Object {
      "marginTop": "20px",
      "width": "100%",
    }
  }
>
  <div
    className="card-body"
  >
    <h2
      className="card-title"
      style={
        Object {
          "color": "#f96302",
        }
      }
    >
      Part #
      mockPartNumber1
    </h2>
    <div
      className="row"
    >
      <div
        className="col-lg-2"
        key="0"
      >
        <label>
          Brand
        </label>
        <div>
          <strong>
            mockBrandName1
          </strong>
        </div>
      </div>
      <div
        className="col-lg-3"
        key="1"
      >
        <label>
          Description
        </label>
      </div>
      <div
        className="col-lg-2"
        key="2"
      >
        <label>
          Supplier
        </label>
        <div>
          <strong>
            mockSupplierName1
          </strong>
        </div>
      </div>
      <div
        className="col-lg-2"
        key="3"
      >
        <label>
          Cost
        </label>
        <div>
          <strong>
            $
            10
          </strong>
        </div>
      </div>
      <div
        className="col-lg-3"
        key="4"
        style={
          Object {
            "textAlign": "right",
          }
        }
      >
        <button
          className="button md"
          id="part-mockPartNumber1"
          key="0"
          onClick={[MockFunction]}
        >
          Edit Part
        </button>
      </div>
    </div>
  </div>
</div>
`;
